src/MICore/Debugger.cs

```diff
@@ -978,24 +999,16 @@ public void OnDebuggerProcessExit(/*OPTIONAL*/ string exitCode)
                         if (_consoleDebuggerInitializeCompletionSource != null)
                         {
                             MIDebuggerInitializeFailedException exception;
-                            string version = GdbVersionFromLog();
-
-                            int majorVersion = -1;
-                            if (!string.IsNullOrWhiteSpace(version))
-                            {
-                                int.TryParse(version.Split('.').FirstOrDefault(), out majorVersion);
-                            }
-                            MICommandFactory.MajorVersion = majorVersion;
-
+                            
                             // We can't use IsMinGW or IsCygwin because we never connected to the debugger
                             bool isMinGWOrCygwin = _launchOptions is LocalLaunchOptions &&
                                     PlatformUtilities.IsWindows() &&
                                     this.MICommandFactory.Mode == MIMode.Gdb;
-                            if (isMinGWOrCygwin && version != null && IsUnsupportedWindowsGdbVersion(version))
+                            if (isMinGWOrCygwin && _gdbVersion != null && IsUnsupportedWindowsGdbVersion(_gdbVersion))
```

Comment:
>_gdbVersion != null [](http://example.com/codeflow?start=51&length=19)

Your code initializes `_gdbVersion` to the empty string, so this will never be true.

Reply:
It can be set to null inside GdbVersionFromLog when _initializationLog == null.